examples/software-device/rs-software-device.cpp

```diff
@@ -0,0 +1,185 @@
+
+#define STB_IMAGE_IMPLEMENTATION
+#include <../third-party/stb_image.h>
+
+using namespace rs2;
+
+struct synthetic_frame
+{
+    int x, y, bpp;
+    std::vector<uint8_t> frame;
+};
+
+synthetic_frame create_synthetic_texture()
+{
+    synthetic_frame frame;
+    auto r = stbi_load_from_memory(splash, (int)splash_size, &frame.x, &frame.y, &frame.bpp, false);
+
+    std::vector<uint8_t> pixels_color(frame.x * frame.y * frame.bpp, 0);
+    
+    memcpy(pixels_color.data(), r, frame.x*frame.y * 4);
+
+    for (auto i = 0; i< frame.y; i++)
+        for (auto j = 0; j < frame.x * 4; j += 4)
+        {
+            if (pixels_color.data()[i*frame.x * 4 + j] == 0)
+            {
+                pixels_color.data()[i*frame.x * 4 + j] = 22;
+                pixels_color.data()[i*frame.x * 4 + j + 1] = 115;
+                pixels_color.data()[i*frame.x * 4 + j + 2] = 185;
+            }
+        }
+    frame.frame = pixels_color;
+
+    return frame;
+}
+
+void fill_synthetic_depth_data(void* data, int w, int h, int bpp, float wave_base)
+{
+    for (int i = 0; i < h; i++)
+    {
+        for (int j = 0; j < w; j++)
+        {
+            auto d = 2 + 0.2 * 0.5 * (1 + sin(wave_base + j / 50.f));
+            ((uint16_t*)data)[i*w + j] = (int)(d * 0xff);
+        }
+    }
+}
+
+int main(int argc, char * argv[]) try
+{
+    const int W = 640;
+    const int H = 480;
+    const int BPP_D = 2;
+
+    rs2_intrinsics depth_intrinsics{ W, H, W / 2, H / 2, W , H , RS2_DISTORTION_BROWN_CONRADY ,{ 0,0,0,0,0 } };
+
+    auto texture = create_synthetic_texture();
+
+    rs2_intrinsics color_intrinsics = { texture.x, texture.y, 
+        (float)texture.x / 2, (float)texture.y / 2,
+        (float)texture.x / 2, (float)texture.y / 2,
+        RS2_DISTORTION_BROWN_CONRADY ,{ 0,0,0,0,0 } };
+
+   
+    //==================================================//
+    //           Declare Software-Only Device           //
+    //==================================================//
+    
+    software_device dev; // Create software-only device
+
+    auto depth_s = dev.add_sensor("Depth"); // Define single sensor 
+    auto color_s = dev.add_sensor("Color"); // Define single sensor 
+
+
+    depth_s.add_video_stream({  RS2_STREAM_DEPTH, 0, 0, 
+                                W, H, 60, BPP_D, 
+                                RS2_FORMAT_Z16, depth_intrinsics });
+
+    depth_s.add_read_only_option(RS2_OPTION_DEPTH_UNITS, 0.001f);
+
+
+    color_s.add_video_stream({  RS2_STREAM_COLOR, 0, 1, texture.x, 
+                                texture.y, 60, texture.bpp, 
+                                RS2_FORMAT_RGBA8, color_intrinsics });
+
+    dev.create_matcher(DLR_C);
+    syncer sync;
+
+    depth_s.start(sync);
+    color_s.start(sync);
+
+    auto depth_stream = depth_s.get_stream_profiles().front();
+    auto color_stream = color_s.get_stream_profiles().front();
+
+    depth_stream.register_extrinsics_to(color_stream, { { 1,0,0,0,1,0,0,0,1 },{ 0,0,0 } });
+
```

Comment:
If not moving the above to a class, please add some noticeable comment here for "End of Software-Only Device Declaration"